portal-impl/src/com/liferay/portal/service/permission/LayoutPermissionImpl.java

```diff
@@ -550,6 +555,12 @@ private boolean _contains(
 				return hasPermission.booleanValue();
 			}
 		}
+		else if (!checkViewableGroup && group.isUserGroup() &&
```

Comment:
This allows any code that tries to filter viewable pages to behave properly with user personal pages that come from user group site templates, because non-administrator users do not have permission to view the user group pages.